src/plinder/data/utils/annotations/ligand_utils.py

```diff
@@ -666,161 +666,171 @@ def annotate_interface_gaps_per_chain(
-        Molecular weight
-    crippen_clogp: float
-        Crippen clogp
-    num_hbd: int
-        Number of hydrogen bonds donor
-    num_hba: int
-        Number of hydrogen bind acceptor
-    num_rings: int
-        Number of rings
-    num_heavy_atoms: int
-        Number of heavy atoms
-    tpsa: float
-        Topological polar surface area
-    qed: float
-        ligand QED score
-    is_covalent: bool = False
-        Is ligand covalently bound to protein
-    covalent_linkages: set[str] = field(default_factory=list)
-        Residue tags of covalently linked residues in ligand
-    neighboring_residues: dict[str, list[int]] = field(default_factory=dict)
-        Dictionary of neighboring residues, with \
-            {instance}.{chain} key and residue number value
-    neighboring_ligands: list[str] = field(default_factory=list)
-        List of neighboring ligands {instance}.{chain}
-    interacting_residues: dict[str, list[int]] = field(default_factory=dict)
-        Dictionary of interacting residues, with \
-            {instance}.{chain} key and residue number value
-    interacting_ligands: list[str] = field(default_factory=list)
-        List of interacting ligands {instance}.{chain}
-    interactions: dict[str, dict[int, list[str]]] = field(
-                    default_factory=dict
-                )
-        Dictionary of {instance}.{chain} to residue number to list of PLIP hashes
-    neighboring_residue_threshold: float = 6.0
-        Maximum distance to consider protein residues neighboring
-    neighboring_ligand_threshold: float = 4.0
-        Maximum distance to consider ligands neighboring
-    num_resolved_heavy_atoms: int | None = None
-        Number of heavy atoms resolved in ligand
-    num_unresolved_heavy_atoms: int | None = None
-        Number of heavy atoms not resolved in ligand
-    is_ion: bool = False
-        Is ligand an ion
-    is_lipinski: bool = False
-        Does ligand satisfy Lipinski Ro5
-    is_fragment: bool = False
-        Does ligand satisfy fragment Ro3
-    is_oligo: bool = False
-        Does ligand match smarts for oligonucleotide, oligosacharide or oligopeptide
-    is_cofactor: bool = False
-        Is ligand in cofactor list
-    in_artifact_list: bool = False
-        Is ligand in artifact list
-    is_artifact: bool = False
-        Is ligand an artifact
-    is_invalid: bool = False
-        Is ligand classified as invalid
-    is_other: bool = False
-        Is ligand classified as other
-    is_invalid: bool = False
-        RDKit invalid
-    posebusters_result: dict[str, ty.Any]
-        Results from running posebusters with "re-dock"
-    unique_ccd_code: str | None = None
-        de-duplicated CCD code
-    waters: dict[str, list[int]]
-        residue numbers and chains of interacting waters
-    residue_numbers: list[int] | None = None
-        residue number(s) of ligand
     """
 
+    def document_ligand_properties_to_tsv(self, filename: str) -> dict[str, ty.Any]:
```

Comment:
this function writes tsv file, eg.
```
L = Ligand()
L.document_ligand_properties_to_tsv('ligand_properties_new.tsv')
```

Known issues
1.  `@cached_property` fields are missing, specifically!
`
{ 'ligand_binding_affinity',
 'ligand_fraction_atoms_with_crystal_contacts',
 'ligand_id',
 'ligand_instance',
 'ligand_is_kinase_inhibitor',
 'ligand_num_atoms_with_crystal_contacts',
 'ligand_num_crystal_contacted_residues',
 'ligand_num_interacting_residues',
 'ligand_num_interactions',
 'ligand_num_neighboring_residues'}
`
Also missing - which is only added at format_ligand functin:
`
'ligand_auth_id',
`

2. datatype is not correctly specified

Reply:
These entries are in our `column_descriptions/ligand_properties.tsv` but I no longer find where they may be added - could be depreciated or added elsewhere:
```
{'ligand_interacting_ligand_chains',
 'ligand_interacting_ligand_chains_BIRD',
 'ligand_interacting_ligand_chains_ECOD',
 'ligand_interacting_ligand_chains_PANTHER',
 'ligand_interacting_ligand_chains_Pfam',
 'ligand_interacting_ligand_chains_UniProt',
 'ligand_interacting_ligand_chains_auth_id',
 'ligand_interacting_ligand_chains_entity_id',
 'ligand_interacting_ligand_chains_length',
 'ligand_interacting_ligand_chains_num_residues'}
 ```

Reply:
@padix-key  - could you have a quick look, if this would make sense?
Also, any ideas about how to pass the datatype specification in a more consistent way?

Reply:
There seems to be an option with `@computed_field` but not sure if that's the best way as such `@cached_property` attributes would still be only accessible at runtime and not independent of data generation:
https://docs.pydantic.dev/2.0/usage/computed_fields

Reply:
@Ninjani mentioned, we tried this before and blew up the memory requirement to load the annotations.